src/Datasource/EntityTrait.php

```diff
@@ -1390,9 +1391,16 @@ public function setSource(string $alias)
      * Returns a string representation of this object in a human readable format.
      *
      * @return string
+     * @deprecated 5.2.0 Casting an entity to string is deprecated. Use json_encode() instead to get a string representation of the entity.
      */
     public function __toString(): string
     {
+        deprecationWarning(
+            '5.2.0',
+            'Casting an entity to string is deprecated. ' .
+            'Use json_encode() instead to get a string representation of the entity.'
+        );
```

Comment:
Should we soft-deprecate in 5.2? I hope this isn't used a lot, but don't know if we're still adding hard-deprecations only near the last minor.

Reply:
I don't mind either way.
@markstory ?

Reply:
I think its fine to add hard deprecations now. Everytime we put it off, we miss adding deprecations :)